apps/backend/src/libs/modules/token/base-token.module.ts

```diff
@@ -0,0 +1,47 @@
+import { type JWTPayload, jwtVerify, SignJWT } from "jose";
+
+import { type Token, type TokenPayload } from "./libs/types/types.js";
+
+type Constructor = {
+	jwtAlgorithm: string;
+	jwtSecret: string;
+	tokenExpirationTime: string;
+};
+
+class BaseToken implements Token {
+	private jwtAlgorithm: string;
+
+	private secret: Uint8Array;
+
+	private tokenExpirationTime: string;
+
+	public constructor({
+		jwtAlgorithm,
+		jwtSecret,
+		tokenExpirationTime,
+	}: Constructor) {
+		this.jwtAlgorithm = jwtAlgorithm;
+		this.secret = new TextEncoder().encode(jwtSecret);
+		this.tokenExpirationTime = tokenExpirationTime;
+	}
+
+	public async create(payload: TokenPayload["payload"]): Promise<string> {
+		const jwt = await new SignJWT({ payload })
+			.setProtectedHeader({ alg: this.jwtAlgorithm })
+			.setIssuedAt()
+			.setExpirationTime(this.tokenExpirationTime)
+			.sign(this.secret);
+
+		return jwt;
+	}
+
+	public async verify<Payload extends JWTPayload = JWTPayload>(
+		jwt: string,
+	): Promise<Payload> {
+		const { payload } = await jwtVerify(jwt, this.secret);
+
+		return payload as Payload;
```

Comment:
I'm curious if there is a posibility to pass it like this instead of casting, if no, then we will leave it this way
```suggestion
		const { payload } = await jwtVerify<Payload>(jwt, this.secret);

		return payload;
```

Reply:
You are right - it works.
Done